### DISCUSSION TITLE
Prediction of Time-To-Event / Understanding the survival function

### DISCUSSION THREAD

**Discussion by nordglueck** [sebp/scikit-survival#432] [Q&A]
Hi there! 
Thanks for your amazing library first of all. 
I'm working on predicting the time-to-event on high dimensional data. I've been using Random Survival Forest on a subset of my features and get a CI of about 0.88. Because this seems to be a very good result I'd like to kind of "manually" verify, that the training and CI is as right as it seems. Therefore I plotted the survival function and I am not sure, if I'm interpreting it right. Therefore I'd like to ask you, what I can read from the plot? 

The corresponding y values would be: 
`array([( True, 2.), ( True, 0.), (False, 2.), (False, 0.), (False, 1.)],
      dtype=[('event', '?'), ('time', '<f8')])`
      
    
<img width="676" alt="image" src="https://github.com/sebp/scikit-survival/assets/62429257/68120a4b-6c66-45bc-85ae-6ee81ff635d4">

And what I'm seeing is, that there is a significant drop in the survival probability of 0 until day 2, which would align with the fact that for 0 the event occurs in 2 days. But for 1 the significant drop would be until day 1, which would differ to the real value of day 0. 

Could you give a little bit more explanation as to how to interpret them? 
Thank you!

**Comment by sebp** [comment]
Your interpretation is correct, except that your plot doesn't explicitly show that the first value of the survival function is always 1.0 for all samples (by definition). Typically, the time point of the first value is at time point zero, but in your data it seems that events can already happen at time 0. Nevertheless, you can see that the curve for sample 1 does have a big initial drop, whereas the other curves do not.
